src/common/image.c

```diff
@@ -2711,6 +2711,17 @@ int dt_image_local_copy_reset(const dt_imgid_t imgid)
 // *******************************************************
 // xmp stuff
 // *******************************************************
+static gboolean _any_altered_data(const dt_imgid_t imgid)
+{
+  if(dt_image_altered(imgid))
+    return TRUE;
+
+  GList *taglist = NULL;
```

Comment:
What about changing dt_tag_get_attached to return only the count when taglist is NULL?

I would call here: `const int count = dt_tag_get_attached(imgid, NULL, TRUE);`

This would avoid the creation if the GList and the immediate release of it which lost quite some CPU cycle.

Reply:
Right. After the weekend...

Reply:
Ok, I may do it if I found some time during the weekend. I'll let you know, you're doing so much at the moment I ought to help a bit here :)

Reply:
BTW, looking at it now it seems better to have another routine as the `dt_tag_get_attached` implementation is already a bit convoluted.

Reply:
Ok, done, see #15379. You should be able to rebase your PR on top of mine.

Reply:
Or you just pick...

Reply:
Right as you prefer.

Reply:
Don't mind at all